src/main/java/com/stripe/net/StripeResponseInterface.java

```diff
@@ -0,0 +1,37 @@
+package com.stripe.net;
+
+import java.time.Instant;
+
+/** Common interface representing an HTTP response from Stripe. */
+interface StripeResponseInterface {
```

Comment:
It seems like the majority of this implementation would be shared right (ie. all the header extraction)? If so, I think we could make some abstract class with the shared implementation to avoid duplicating the code and have both implementations extend this instead of (or along with) the interface. This might be nice for testing as well.